docs/IPFS.md

```diff
@@ -0,0 +1,67 @@
+
+## cURL handling of the IPFS protocols
+The IPFS integration in cURL hides this gateway logic for you. So instead of providing a full url to a file on IPFS like this:
+```
+curl http://127.0.0.1:8080/ipfs/bafybeigagd5nmnn2iys2f3doro7ydrevyr2mzarwidgadawmamiteydbzi
+```
+
+You can provide it with the IPFS protocol instead:
+```
+curl ipfs://bafybeigagd5nmnn2iys2f3doro7ydrevyr2mzarwidgadawmamiteydbzi
+```
+
+With the IPFS protocol way of asking a file, cURL still needs to know the gateway. cURL essentially just rewrites the IPFS based URL to a gateway URL.
+
+### IPFS_GATEWAY environment variable
+If the `IPFS_GATEWAY` environment variable is found, it's value is used as gateway.
+
+### Automatic gateway detection
+When you provide no additional details to cURL then cURL will:
+
+1. First look for the `IPFS_GATEWAY` environment variable and use that if it's set.
+2. Look for the file: `~/.ipfs/gateway`. If it can find that file then it means that you have a local gateway running and that file contains the URL to your local gateway.
+
+If cURL fails you'll be presented with an error message and a link to this page to the option most applicable to solving the issue.
+
+### --ipfs-gateway argument
+You can also provide a `--ipfs-gateway` argument to cURL. This overrules any other gateway setting. cURL won't fallback to the other options if the provided gateway didn't work.
+
+## Gateway redirects
+A gateway could redirect to another place. For example, dweb.link redirects [path based](https://docs.ipfs.tech/how-to/address-ipfs-on-web/#path-gateway) requests to [subdomain based](https://docs.ipfs.tech/how-to/address-ipfs-on-web/#subdomain-gateway) ones. So a request to:
+```
+curl ipfs://bafybeigagd5nmnn2iys2f3doro7ydrevyr2mzarwidgadawmamiteydbzi --ipfs-gateway https://dweb.link
+```
+Which would be translated to:
+```
+https://dweb.link/ipfs/bafybeigagd5nmnn2iys2f3doro7ydrevyr2mzarwidgadawmamiteydbzi
+```
+Will redirect to:
+```
+https://bafybeigagd5nmnn2iys2f3doro7ydrevyr2mzarwidgadawmamiteydbzi.ipfs.dweb.link
+```
+If you thrust this behavior from your gateway of choice then passing the `-L` option will follow the redirect.
+
+## Error messages and hints
+Depending on the arguments, cURL could present the user with an error.
+
+### Gateway file and environment variable
+cURL tried to look for the file: `~/.ipfs/gateway` but couldn't find it. It also tried to look for the `IPFS_GATEWAY` environment variable but couldn't find that either. This happens when no extra arguments are passed to cURL and letting it try to figure it out [automatically](#Automatic-gateway-detection).
+
+You could fix this a couple different ways. The easiest is to run a [KUBO node](https://github.com/ipfs/kubo/releases) of version 0.15.0 or higher. Running an up to date version of [IPFS Desktop](https://github.com/ipfs/ipfs-desktop/releases) should fix it too.
```

Comment:
This feels like recommendations that will grow old very fast. Can you rephrase this without mentioning any specific product or version number? The advice is to run an IPFS gateway of your own isn't it? It is not 100% clear to me that "KUBO" and "IPFS Desktop" are examples of such.

But also: just because curl doesn't find the file or the environment variable does not mean that there is no gateway running. It is more like forgetting to put the sign up where to find it. Isn't it?